src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -337,6 +337,11 @@ private void ProcessExpressionParameter()
             if (ExcludeProperty != null)
             {
                 _exclusionFilter = new MshExpressionFilter(ExcludeProperty);
+                if ((Property == null) || (Property.Length == 0))
+                {
```

Comment:
The conclusion is to take your changes to `ProcessExpressionParameter()` in your second commit -- always add "*" to `-Property` when `-ExcludeProperty` is specified but `-Property` is not : https://github.com/PowerShell/PowerShell/pull/2421/commits/70c30384c74d9f6392b13cc9d04baf9eb5f8c364

I think you implemented the pseudo code I listed above to avoid the breaking change, but as we discussed here, that will raise new confusion about how `-ExlcudeProperty` works. So I believe it's reasonable to take your original changes to `ProcessExpressionParameter()` to make it simple for users to understand.